src/opcode.rs

```diff
@@ -42,20 +47,67 @@ impl<DB> Inspector<DB> for OpcodeCounterInspector
 where
     DB: Database,
 {
-    fn step(&mut self, interp: &mut Interpreter, _context: &mut EvmContext<DB>) {
+    fn step(&mut self, interp: &mut Interpreter, context: &mut EvmContext<DB>) {
+        self.initialize_interp(interp, context);
+
         let opcode_value = interp.current_opcode();
         if let Some(opcode) = OpCode::new(opcode_value) {
             *self.opcode_counts.entry(opcode).or_insert(0) += 1;
+        }
 
-            let gas_table =
-                revm::interpreter::instructions::opcode::spec_opcode_gas(_context.spec_id());
-            let opcode_gas_info = gas_table[opcode_value as usize];
+        self.step_end(interp, context);
+        let gas_table = revm::interpreter::instructions::opcode::spec_opcode_gas(context.spec_id());
+        let opcode_gas_info = gas_table[opcode_value as usize];
 
-            let opcode_gas_cost = opcode_gas_info.get_gas() as u64;
+        let opcode_gas_cost = opcode_gas_info.get_gas() as u64;
 
-            *self.opcode_gas.entry(opcode).or_insert(0) += opcode_gas_cost;
+        if let Some(opcode) = OpCode::new(opcode_value) {
+            let gas_used = self.gas_inspector.last_gas_cost();
```

Comment:
this is a bit tricky because gas cost track cost of the entire call so far

we need to track gas at the beginning of the step and on step end we can get the diff by comparing `.remaining`

see also TracingInspector for reference

Reply:
> this is a bit tricky because gas cost track cost of the entire call so far
> 
> we need to track gas at the beginning of the step and on step end we can get the diff by comparing `.remaining`
> 
> see also TracingInspector for reference

hey thank you for the review, i'm not sure i understand this